src/listener.ts

```diff
@@ -51,10 +51,19 @@ const responseViaCache = (
 
 const responseViaResponseObject = async (
   res: Response | Promise<Response>,
-  outgoing: ServerResponse | Http2ServerResponse
+  outgoing: ServerResponse | Http2ServerResponse,
+  customErrorHandler?: (e: unknown) => void
```

Comment:
How about making the argument an object instead of a `customErrorHandler`? 

```ts
const responseViaResponseObject = async (
  res: Response | Promise<Response>,
  outgoing: ServerResponse | Http2ServerResponse,
  options?: {
    errorHandler?: (e: unknown) => void
  }
) => {
```

In that case, I would prefer a simple `errorHandler` in the `options`.